orttraining/orttraining/training_api/include/lr_scheduler.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "orttraining/training_api/include/optimizer.h"
+
+namespace onnxruntime {
+namespace training {
+namespace api {
+
+/**
+ * @brief Base class for learning rate scheduler.
+ */
+struct LRSchedulerBase {
+ public:
+  LRSchedulerBase(std::shared_ptr<Optimizer> optimizer) : optim_(optimizer) {}
```

Comment:
The optimizer is owned by the `TrainingSession`. If the `LRScheduler` is needed, it must also have access to the optimizer. Either we use shared ownership in this case, or the `LRScheduler` must hold onto a raw pointer/reference to the optimizer.
Since the ownership seems to be shared between the `LRScheduler` and the user's model script/`TrainingSession`, we chose to use the `shared_ptr`.